include/support.h

```diff
@@ -159,4 +160,28 @@ bool ends_with(const std::string &suffix, const std::string &str) noexcept;
 
 bool is_executable_filename(const std::string &filename) noexcept;
 
+// extremely fast sine and cosine approximations good to roughly four digits
+// of accuracy (suitable for 16-bit audio calculations).
+// Credit: Bhaskara I (c. 600 – c. 680) sine approximation formula
+constexpr float fast_sin(float rad)
```

Comment:
Do we have any benchmarks suggesting this fast_sin implementation is indeed faster than library provided `sinf`?

I'm asking because providing numerical functions is actually quite a big topic, and using this algorithm is rather unheard of when implementing sine/cosine functions… also, I don't know what are numerical properties of this solution. Generally, I think Taylor series expansion would probably be faster due to no branching in the code… but for low accuracy like this, very simple lookup table might also be very good.

Picking the right algorithm here is really *deep* topic - my concern here - do we really get any benefit?

Some interesting discussions on SO touching this topic:

- https://stackoverflow.com/a/18662397
- https://stackoverflow.com/a/37947583
- https://stackoverflow.com/a/345117

(but there's whole host of literature, research papers, etc about this)

Also, it might be the case that we could get faster solution by using `-ffast-math` (which we already do sometimes IIRC?) and maybe `-mfpath=sse`?

Reply:
Answered below :-)  A couple points:
- `sinf` and `cosf` performed the same as `sin` and `cos`
- `-ffast-math` is implied by `Ofast`, which I used in my tests
- I enabled all hardware operations, so the compiler was free to use its fastest `sin` implementation (AVX, SSE2 on x64, NEON on ARM, and AltiVec on PPC).